src/code/code_800C3C20.c

```diff
@@ -5,9 +5,9 @@ u8 sSfxBankIds[] = {
 };
 
 void AudioMgr_StopAllSfx(void) {
-    u32 i;
+    volatile u8* bankIdPtr;
 
-    for (i = 0; (u32)(i < ARRAY_COUNT(sSfxBankIds)); i++) {
-        Audio_StopSfxByBank(sSfxBankIds[i]);
+    for (bankIdPtr = &sSfxBankIds[0]; bankIdPtr < (sSfxBankIds + ARRAY_COUNT(sSfxBankIds)); bankIdPtr++) {
```

Comment:
```suggestion
    for (bankIdPtr = &sSfxBankIds[0]; bankIdPtr < (sSfxBankIds + sizeof(sSfxBankIds)); bankIdPtr++) {
```
Since this is pointer math, I think `sizeof` makes more sense. `ARRAY_COUNT` only matches because `sSfxBankIds` is size `u8`. If `sSfxBankIds` was switched to a `u32`, `ARRAY_COUNT` would break but `sizeof` would still match

Reply:
It should still work. Pointer arithmetic includes the size of the underlying element and adds accordingly. This is why pointer iterators work with just `++`.

Reply:
`sizeof` actually gives the wrong result when changing the type.
For example with sizeof at a type of `u16` I get that the end of the array would be at `sSfxBankIds+0x1c` which is 14 `u16` elements.
If I use ARRAY_COUNT I get `sSfxBankIds+0xe` which is the same 7 `u16` elements.

Reply:
Oh right yeah you're right